core/src/loader.rs

```diff
@@ -410,46 +487,82 @@ impl<'gc> Loader<'gc> {
+                    match sniffed_type.expect_known()? {
+                        ContentType::Swf => {
+                            let movie = Arc::new(SwfMovie::from_data(
+                                &data,
+                                Some(url.into_owned()),
+                                loader_url,
+                            )?);
+
+                            let mut activation = Avm2Activation::from_nothing(uc.reborrow());
+                            let parent_domain = activation.avm2().global_domain();
+                            let domain = Avm2Domain::movie_domain(&mut activation, parent_domain);
                             uc.library
                                 .library_for_movie_mut(movie.clone())
-                                .register_character(
-                                    id,
-                                    crate::character::Character::MorphShape(morph_shape),
+                                .set_avm2_domain(domain);
+
+                            if let Some(broadcaster) = broadcaster {
+                                Avm1::run_stack_frame_for_method(
+                                    clip,
+                                    broadcaster,
+                                    NEWEST_PLAYER_VERSION,
+                                    uc,
+                                    "broadcastMessage".into(),
+                                    &[
+                                        "onLoadProgress".into(),
+                                        clip.object(),
+                                        data.len().into(),
+                                        data.len().into(),
+                                    ],
                                 );
+                            }
+
+                            if let Some(mut mc) = clip.as_movie_clip() {
+                                mc.replace_with_movie(uc.gc_context, Some(movie.clone()));
+                                mc.post_instantiation(uc, None, Instantiator::Movie, false);
+
+                                let mut morph_shapes = fnv::FnvHashMap::default();
+                                mc.preload(uc, &mut morph_shapes);
+
+                                // Finalize morph shapes.
+                                for (id, static_data) in morph_shapes {
+                                    let morph_shape = MorphShape::new(uc.gc_context, static_data);
+                                    uc.library
+                                        .library_for_movie_mut(movie.clone())
+                                        .register_character(
+                                            id,
+                                            crate::character::Character::MorphShape(morph_shape),
+                                        );
+                                }
+                            }
                         }
+                        ContentType::Gif | ContentType::Jpeg | ContentType::Png => {
```

Comment:
TODO for future PR: I believe this is actually version-gated, v7 SWFs can not load PNGs, for example. Not sure about the specifics, but worth a comment.

Reply:
The exact specifics are that only `ContentType::Jpeg` works on v7. Unfortunately I can't build tests for it as the output is identical in both cases; one just doesn't render the image.

I have to wonder if this also applies to `DefineBitsJpeg2` tags in SWFs.